Calculator.c

```diff
@@ -0,0 +1,101 @@
+#include<stdio.h>
+#include<stdlib.h>
+#include<ctype.h>
+
+int main()
+{
+    char s[100];
+    printf("Enter the expression for evaluation: ");
+    scanf("%[^\n]",s);
+
+    int arr_num[100],index_num=0,index_op=0;
+    char arr_op[100];
+    int digit=0;
+    int i,j,k;
+    for(i=0;s[i]!='\0';i++)
+    {
+        char ch=s[i];
+        if(isspace(ch))
+        {
+            continue;
+        }
+        if(isdigit(ch))
+        {
+            digit=digit*10+(ch-'0');
+        }
+        else
+        {
+            arr_num[index_num++]=digit;
+            digit=0;
+            if(ch=='+'||ch=='-'||ch=='/'||ch=='*')
+            {
+                arr_op[index_op++]=ch;
+            }
+            else
+            {
+                printf("Error: Invalid expression.\n");
+                return 1;
+            }
+        }
+    }
+    arr_num[index_num++]=digit;
+
+    for(i=0;i<index_op;i++)
```

Comment:
Are these many loops really required to solve this problem?
From line no 43 to 65 and 67 to 84, most of the code is repetitive? 
Do we need to write separate code for each operator?